app/models/miq_worker/runner.rb

```diff
@@ -374,6 +375,32 @@ def heartbeat_to_drb
   def heartbeat_to_file(timeout = nil)
     timeout ||= worker_settings[:heartbeat_timeout] || Workers::MiqDefaults.heartbeat_timeout
     File.write(@worker.heartbeat_file, (Time.now.utc + timeout).to_s)
+
+    get_messages.each { |msg, *args| process_message(msg, *args) }
```

Comment:
For this PR, I was just looking to get to functional equivalence with messaging through Drb. We can start evolving this in future PRs as we start removing Drb from the workers.